src/commonMain/kotlin/fr/acinq/lightning/channel/Commitments.kt

```diff
@@ -236,6 +236,13 @@ data class Commitment(
 
     fun isIdle(changes: CommitmentChanges): Boolean = hasNoPendingHtlcs() && changes.localChanges.all.isEmpty() && changes.remoteChanges.all.isEmpty()
 
+    /** The commitment is locked once we've sent and received splice_locked. */
+    fun ChannelContext.isLocked(): Boolean {
+        val localLocked = staticParams.useZeroConf || localFundingStatus is LocalFundingStatus.ConfirmedFundingTx
+        val remoteLocked = remoteFundingStatus == RemoteFundingStatus.Locked
+        return localLocked && remoteLocked
+    }
+
```

Comment:
This isn't equivalent to the code that was in `deactivateCommitments` and I think is incorrect. In the case where there are several funding tx chained, we may not receive a splice_locked for every single one and need to handle the intersection between local and remote.

Reply:
Good catch! But there was an issue in `deactivateCommitments` (or at least something that bothered me):

```kotlin
val lastLocalLockedIndex: Long = active.firstOrNull { staticParams.useZeroConf || it.localFundingStatus is LocalFundingStatus.ConfirmedFundingTx }?.fundingTxIndex ?: -1
val lastRemoteLockedIndex: Long = active.firstOrNull { it.remoteFundingStatus is RemoteFundingStatus.Locked }?.fundingTxIndex ?: -1
val lastLockedIndex = min(lastLocalLockedIndex, lastRemoteLockedIndex)
return when (val lastLocked = active.find { it.fundingTxIndex == lastLockedIndex }) {
```

The `lastLocked` commitment could be a different commitment from the one we use to compute `lastLocalLockedIndex` and `lastRemoteLockedIndex`, if we have RBF attempts, and I don't think this is correct...but that can probably be fixed differently, I'll take another look at this.

Reply:
Fixed in https://github.com/ACINQ/lightning-kmp/pull/438/commits/6fb40021dd021b1ce9a9e1651a8a54224ff9ca90

Reply:
I think it's still wrong, because we are not handling the intersection. There has to be a `min` somewhere because we want the "common denominator".

Reply:
I also added tests for this behavior in https://github.com/ACINQ/lightning-kmp/pull/438/commits/82ca8f5556ac46d8249909c9b34f35a0140b0fce

Reply:
Very nice! I verified that the test indeed fails with the previous `deactivateCommitments`.